fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java

```diff
@@ -1,61 +1,198 @@
+        assertCompartmentSearchDoesntReturnComposition(DEVICE, PATIENT_ID, "subject:Basic.integer", "12");
+    }
+
+    @Test
+    public void testSearchPatientCompartment_relativeReference_chained_quantity() throws Exception {
+        assertCompartmentSearchReturnsComposition(PATIENT, PATIENT_ID, "subject:Basic.Quantity", "25|http://unitsofmeasure.org|s");
+        assertCompartmentSearchReturnsComposition(PRACTITIONER, PRACTITIONER_ID, "subject:Basic.Quantity", "25|http://unitsofmeasure.org|s");
+        assertCompartmentSearchDoesntReturnComposition(PATIENT, OTHER_ID, "subject:Basic.Quantity", "25|http://unitsofmeasure.org|s");
+        assertCompartmentSearchDoesntReturnComposition(DEVICE, PATIENT_ID, "subject:Basic.Quantity", "25|http://unitsofmeasure.org|s");
     }
 
-    @BeforeClass
-    public void createReference() throws FHIRException {
-        String originalRequestUri = "https://example.com/Patient/123";
-        FHIRRequestContext context = FHIRRequestContext.get();
-        context.setOriginalRequestUri(originalRequestUri);
+    @Test
+    public void testSearchPatientCompartment_relativeReference_chained_reference() throws Exception {
+        // Note that "Reference just happens to be another searchable parameter in the Basic resource.
+        assertCompartmentSearchReturnsComposition(PATIENT, PATIENT_ID, "subject:Basic.Reference", "Patient/123");
+        assertCompartmentSearchReturnsComposition(PRACTITIONER, PRACTITIONER_ID, "subject:Basic.Reference", "Patient/123");
+        assertCompartmentSearchDoesntReturnComposition(PATIENT, OTHER_ID, "subject:Basic.Reference", "Patient/123");
+        assertCompartmentSearchDoesntReturnComposition(DEVICE, PATIENT_ID, "subject:Basic.Reference", "Patient/123");
+    }
+
+    @Test
+    public void testSearchPatientCompartment_relativeReference_chained_string() throws Exception {
+        assertCompartmentSearchReturnsComposition(PATIENT, PATIENT_ID, "subject:Basic.string", "testString");
+        assertCompartmentSearchReturnsComposition(PRACTITIONER, PRACTITIONER_ID, "subject:Basic.string", "testString");
+        assertCompartmentSearchDoesntReturnComposition(PATIENT, OTHER_ID, "subject:Basic.string", "testString");
+        assertCompartmentSearchDoesntReturnComposition(DEVICE, PATIENT_ID, "subject:Basic.string", "testString");
+    }
+
+    @Test
+    public void testSearchPatientCompartment_relativeReference_chained_token() throws Exception {
+        assertCompartmentSearchReturnsComposition(PATIENT, PATIENT_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
+        assertCompartmentSearchReturnsComposition(PRACTITIONER, PRACTITIONER_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
+        assertCompartmentSearchDoesntReturnSavedResource(PATIENT, OTHER_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
```

Comment:
Should this be assertCompartmentSearchDoesntReturnComposition?

Reply:
drat, I thought I got all the places...definitely missed this line (and the following one)